scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.jsx

```diff
@@ -106,9 +105,7 @@ SpellcheckerContextMenuComponent.propTypes = {
  * @description Maps the values from state to the component callback type props.
  */
 const mapDispatchToProps = (dispatch) => ({
-    addWord: (word) => dispatch(actions.ignoreWord(word)),
-    ignoreWord: (word) => dispatch(actions.addWord(word)),
-    replaceWord: (word) => setTimeout(() => dispatch(actions.replaceWord(word)), 0),
+    replaceWord: (word) => setTimeout(() => dispatch(actions.replaceWord(word)), 0)
```

Comment:
Instead, we could change `spellcheck.addWord` to return [this line](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/core/spellcheck/spellcheck.js#L372), which is already a promise if we look at the definition [here](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/apps/dictionaries/services/DictionaryService.js#L175) and only trigger the event after the promise is completed.

This way you can get rid of the timeout.